### ISSUE TITLE
How to add additional parameter to objective function?

### ISSUE THREAD

**Issue by igor-susic** [tensorflow/tfjs#2796]
Is it possible to add any other parameter to the objective function besides `y_true` and `y_pred`? If it is, how would one do it? I can't find appropriate documentation or answer anywhere ...

**Comment by rthadur** [comment]
@igor-susic can you please provide link to the function ?

**Comment by igor-susic** [comment]
@rthadur Sorry for the delayed answer. 

So, if you look at the reinforcement learning, and algorithms like `PPO` the loss function looks like this:

![PPO Clip loss](https://spinningup.openai.com/en/latest/_images/math/99621d5bcaccd056d6ca3aeb48a27bf8cc0e640c.svg)

You need much more than just `y_pred` and `y_true`, for now, I have found a simple way to do this, but I don't like using global variables and having to keep tabs on them.

Here is my solution:

If you, for example, use Adam optimizer:

```javascript
model.compile(
   {
        'optimizer': tf.train.adam(1e-4),
        'loss': [this.computeLoss().bind(this)],
   }
);
```
where `computeLoss` is defined like:

```javascript
computeLoss() {
        const loss = (y_true, y_pred) => {
            // Here you would use some variables outside the function
           // this.rewards etc. 
        };
    return loss;
```

I know that I could use `CORE API` but then the whole process is more bug-prone and harder to
code and understand.

Is there any way that I could use some other variables that could maybe be passed to the model as input like in [Keras](https://github.com/keras-team/keras/issues/2121)?

**Comment by tafsiri** [comment]
You can use optimizer.minimize or optimizer.applyGradients directly. Here is an example of a DQN with custom loss function, https://github.com/tensorflow/tfjs-examples/blob/master/snake-dqn/agent.js#L133

**Comment by rthadur** [comment]
Closing this due to lack of activity, feel to reopen. Thank you
